### DISCUSSION TITLE
Recommended way to package a Python program with dependencies?

### DISCUSSION THREAD

**Discussion by jenny923432** [ScoopInstaller/Scoop#5991] [Help and Questions]
Hoping someone can help me out with a guide or some tips here because I've been trying for a while and can't figure a good solution out. It must be a common thing, but I couldn't find much info on Google either (it's hard because there's a different Python library also called Scoop).

If you've got a simple Python program that depends on some pip packages, what's the best way to package and distribute it via Scoop?

Do you ship just the .py files and specify a Python dependency with a minimum version? And if so, how do you get ahold of that dependency executable to run your program and install the pip packages? 

Or do you ship a virtual environment with a Python and the pip packages already included? This could be wasteful, but it could also avoid the issue of figuring out whether to uninstall those pip packages when your Scoop package is uninstalled... And if you do ship use a venv, how do make the paths in it non-absolute -- create the venv during the post-install script?

**Comment by HUMORCE** [comment]
- Create venv while installing, as you said.
- Bundle a minimal python environment.
- Build executables with Nuitka, PyInstaller, etc., if it's your project.
